src/qibolab/platforms/multiqubit.py

```diff
@@ -279,12 +287,22 @@ def disconnect(self):
                 self.instruments[name].disconnect()
             self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence: PulseSequence, nshots=None):
+    def execute_pulse_sequence(self, sequence, options, **kwargs):
         if not self.is_connected:
             raise_error(RuntimeError, "Execution failed because instruments are not connected.")
-        if nshots is None:
+
+        if options.averaging_mode == AveragingMode.SINGLESHOT:
+            nshots = 1
+            self.average = False
+        elif options.averaging_mode == AveragingMode.CYCLIC:
+            nshots = options.nshots
+            self.average = True
+
+        if options.nshots is None:
             nshots = self.nshots
```

Comment:
`options.nshots` will never be `None` at least on the new platform

Reply:
In the current qibocal main branch that is not true. For example:

![imagen](https://github.com/qiboteam/qibolab/assets/39095664/76dc9d3d-97d4-4d0b-b898-4c29a85d54a0)

The defaukt value of nshopts is None, so if nshots is not defined in the action runcard, the params.nshots could be None.

Reply:
That's not a problem. `ExecutionParameters` would get the nshots value defined on the runcard as default.

Reply:
it is not getting them... If I remove the init, the value is None.

Reply:
From the platform

```
def execute_pulse_sequence(self, sequence, options, **kwargs):
        """Executes a pulse sequence.

        Args:
            sequence (:class:`qibolab.pulses.PulseSequence`): Pulse sequence to execute.
            options (:class:`qibolab.platforms.platform.ExecutionParameters`): Object holding the execution options.
            **kwargs: May need them for something

        Returns:
            Readout results acquired by after execution.
        """
        if options.nshots is None:
            options = replace(options, nshots=self.nshots)
```

Reply:
The point is that when we create the platform for qblox, we create Multiqubit platform, and the execute_pulse_sequence method executed is the one of the class Multiqubit, that is why we need to adde those lines also in Multiqubit.